index.bs

```diff
@@ -954,17 +883,72 @@ interface FileReader: EventTarget {
   attribute EventHandler onabort;
   attribute EventHandler onerror;
   attribute EventHandler onloadend;
-
 };
 </pre>
 
+<div algorithm="read operation">
+A {{FileReader}} has an associated <dfn id=readOperation>read operation</dfn> algorithm,
+given |blob|, a |type| and an optional |encodingName|,
+runs the following steps:
+
+1. Set the [=context object=]'s {{FileReader/result}} to `null`.
+1. Set the [=context object=]'s {{FileReader/error!!attribute}} to `null`.
```

Comment:
Setting readonly attributes is not great; this should set an internal slot/associated value which those getters return the value of.

Reply:
Done (and in doing so, realized that I somehow had lost the step that sets readyState to loading. Fixed that as well).